pkg/kn/commands/config/channel_type_mapping.go

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2022 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configCmd
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/commands/config/channel"
+	"knative.dev/client/pkg/kn/config"
+)
+
+var defaultChannelTypeUpdater = viperUpdater(viper.GetViper(), config.KeyChannelTypeMappings)
+
+// NewConfigChannelTypeMappingCommand represents 'kn config channel-type-mapping' command
+func NewConfigChannelTypeMappingCommand(p *commands.KnParams) *cobra.Command {
+	channelTypeMappingCmd := &cobra.Command{
+		Use:   "channel-type-mapping",
+		Short: "Manage channel type mapping list",
+	}
+	channelTypeMappingCmd.AddCommand(newChannelTypeDeleteCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeAddCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeUpdateCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeListCommand(p))
+	return channelTypeMappingCmd
+}
+
+func addChannelTypeMappingFlags(flagSet *pflag.FlagSet, channelType *config.ChannelTypeMapping) {
+	flagSet.StringVar(&channelType.Kind, "kind", "", "The kind of Channel, such as InMemoryChannel or KafkaChannel, based on the default ConfigMap")
+	flagSet.StringVar(&channelType.Group, "group", "", "The API group of the Kubernetes resource")
+	flagSet.StringVar(&channelType.Version, "version", "", "The version of the Kubernetes resource")
+}
+
+func newChannelTypeUpdateCommand(_ *commands.KnParams) *cobra.Command {
+	channelType := config.ChannelTypeMapping{}
+	updateChannelTypeMapCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update channel type mapping by its alias",
+		Args:  cobra.ExactArgs(1),
+		RunE:  modifyList(channel.Update(&channelType), defaultChannelTypeUpdater),
+	}
+	addChannelTypeMappingFlags(updateChannelTypeMapCmd.Flags(), &channelType)
+	return updateChannelTypeMapCmd
+}
```

Comment:
Wrt Max's and my previous comment. This whole function should move to `./channel/update.go` and then 

And then L36 would change to something like:
```
...AddCommand(NewChannelTypeUpdateCommand(p))
```
In addition I'd not hesitate to move whole `channel_type_mapping.go` to `channel` dir/package. In other words to move sub-command to its own package together with its actions.
E.g. see the structure of `kn broker` group https://github.com/knative/client/tree/main/pkg/kn/commands/broker.

Reply:
I see. No problem.